### DISCUSSION TITLE
ESP OTA update required

### DISCUSSION THREAD

**Discussion by virus100b** [Dilbert66/esphome-dsckeybus#134] [General]
With the last update of ESPHome the OTA: need to be updated in the code with the new format

ota:
  - platform: esphome
    password: !secret ota_password

@Dilbert66  what about the below lines. Do we still need them?

   safe_mode: True
   on_begin:
       switch.turn_off: connection_status_switch

**Comment by Dilbert66** [comment]
Move "safe_mode:" to it's own section as per the Esphome documentation. The rest you can leave as is.

Edit: For now there is a bug in the newest esphome when using an on_begin: section so you can comment it out with a # sign or just delete the two lines.  The switch turn off statement is used to turn off the dsc process interrupts as they could potentially cause ota issues but in my experience this was more an issue with the esp8266. With the ESP32, i have not seen problems.

**Comment by macentrasher** [comment]
This is what i did

ota:
  platform: esphome
  password: !secret ota_password
  on_begin:
    then:
      - switch.turn_off: connection_status_switch

safe_mode:
  on_safe_mode:
    then:
      - switch.turn_off: connection_status_switch

**Reply by virus100b** [reply]
i just update like this and is fine 

ota:
  - platform: esphome
    password: !secret ota_password

Do we need safe mode?

**Reply by macentrasher** [reply]
safe mode will allow you to update ota in case the ESP gets stuck in a boot loop. After 10 it will go into safe mode. When in safe mode you can still update OTA without having to pull your board out of your alarm box. You dont need it but its a nice to have. That Yaml I pasted above was the only thing I needed to change.

**Reply by virus100b** [reply]
Ok. Thank you for info. I will add that too.
